### PR TITLE
Add :no_conflict configuration option

### PR DESCRIPTION
**Pull request by dividedharmony**
This PR adds `Spree::RelatedProductsConfiguration` class to allow for host app to pass in a `:no_conflict` configuration option.

If the `:no_conflict` configuration is set to `true`, then the method_missing defined relations in the product and variant decorators are disabled. This is to appease developers that think it is dangerous to allow end-users to define metaprogrammatic behavior in their application.

By default, `:no_conflict` is set to `false` so as not to break any existing apps.

### CODE REVIEW

File: app/decorators/models/solidus_related_products/product/add_relation_methods.rb [hunk 1]
```diff
@@ -72,6 +72,7 @@ def relation_filter_for_relation_type(relation_type)
       # it up the inheritance chain.
       # rubocop:disable Style/MissingRespondToMissing
       def method_missing(method, *args)
```
Comment (dividedharmony) [inline app/decorators/models/solidus_related_products/product/add_relation_methods.rb:74]: Originally, I was going to skip the `method_missing` definition altogther if `:no_conflict` were true. Like so:
```rb
unless ::SolidusRelatedProducts::Config[:no_conflict]
  def method_missing(method, *args)
    # ...
  end
end
```
However, I found that this way was much more testable.

File: app/decorators/models/solidus_related_products/variant/add_relation_methods.rb [hunk 2]
```diff
@@ -52,6 +52,7 @@ def relation_filter_for_relation_type(relation_type)
       # it up the inheritance chain.
       # rubocop:disable Style/MissingRespondToMissing
       def method_missing(method, *args)
```
Comment (dividedharmony) [inline app/decorators/models/solidus_related_products/variant/add_relation_methods.rb:54]: Ditto for variant's `method_missing` :point_up:

File: app/decorators/models/solidus_related_products/product/add_relation_methods.rb (hunk 1 above)
Comment (tvdeyen) [reply app/decorators/models/solidus_related_products/product/add_relation_methods.rb:74]: Agreed

File: app/decorators/models/solidus_related_products/variant/add_relation_methods.rb (hunk 2 above)
Comment (tvdeyen) [reply app/decorators/models/solidus_related_products/variant/add_relation_methods.rb:54]: 👍🏻

**Review by tvdeyen — COMMENTED** [review]
Nice work. I have one non-code related question. Why is this not real fork?

A real fork would give us some advantages, like CI and the ability to open PRs to the upstream project.

**Comment by dividedharmony** [comment]
@tvdeyen Sorry, I forgot about the Fork button. I'll open a new repo that is a direct Fork of the upstream project.

**Pull request closed without merging**